scene/gui/tree.cpp

```diff
@@ -2385,8 +2386,8 @@ bool Tree::edit_selected() {
 		return true;
 
 	} else if (c.mode == TreeItem::CELL_MODE_STRING || c.mode == TreeItem::CELL_MODE_RANGE || c.mode == TreeItem::CELL_MODE_RANGE_EXPRESSION) {
-
-		Point2i textedpos = get_global_pos() + rect.pos;
+		Vector2 ofs(0, (text_editor->get_size().height - rect.size.height) / 2);
+		Point2i textedpos = get_global_pos() + rect.pos - ofs;
```

Comment:
another way to consider would be to adapt the style box to fit.
```cpp
style = text_editor->get_stylebox("normal")->duplicate();
style->set_margin(MARGIN_TOP /*do some math with tree cell height (TreeItem height) and font size to set top and bottom margin*/);
text_edtitor->add_style_override("normal", style);
```
than there would be no overlapping... and the item always would be correct size.

Reply:
Thinking about it a little more...
maybe best actually would be to give control over the style of the lineEdit.
This would allow to adapt it to whatever the designer desires + can be adapted to all the other tree settings.
so keeping you changes since the centering is already a big advantage + adding a style box to Tree for the line edit.
`theme->set_stylebox("line_edit", "Tree", line_edit_with_adapted_margins);`
this would allow most flexibility.

Reply:
than you could give the style you posted and fixed it to the tree and the style for the normal line edit won't change

Reply:
yes. I agree with having a LineEdit stylebox for Tree would be better.
I'd like to have it on 3.0